fables/parse.py

```diff
@@ -83,6 +83,13 @@ def remove_data_before_header(df: pd.DataFrame, force_numeric: bool) -> pd.DataF
 
 
 def post_process_dataframe(df: pd.DataFrame, force_numeric: bool) -> pd.DataFrame:
+    # Remove columns that have no header and have only null data.
+    for col in df.columns:
+        if (pd.isnull(col) or str(col).startswith("Unnamed: ")) and len(
+            df[col].value_counts()
+        ) == 0:
```

Comment:
I know this code was only moved, but since it is always applied rather than only when `if num_rows_before` I had a performance double take.

I'm not 100% sure if they are equivalent, but `... and df[col].isnull().all()` may be faster than `... and len(df[col].value_counts()) == 0` in the best case since it will exit early when it finds the first non-null value.

I think it could be worth trying this optimization to see if the test still passes.

Reply:
Thanks, I'll give it a go. 😄

Reply:
Appears to accomplish the same, test still passes, and validated on test files.